src/rank_llm/rerank/reranker.py

```diff
@@ -134,7 +134,7 @@ def write_rerank_results(
 
         name = self._model_coordinator.get_output_filename(
             top_k_candidates, dataset_name, shuffle_candidates, **kwargs
```

Comment:
Please change it within `get_output_filename` functions in all the rerankers in case they are/will be called from other places.
If you define a helper function (`get_formatted_datetime` or something similar) in the root RankLLM class, then all the `get_output_filename` implementations can call that.

Please also do a git grep to find datetime to see if we have this problem anywhere else, since ":" in the name comes from the datetime.isoformat that we append to the output files.